src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,633 @@
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumerationArg {
+    /// A human-readable name for this argument to help clarify use.
+    name: Option<String>,
+
+    /// Additional description for this argument.
+    description: Option<String>,
+
+    /// List of valid string options for this argument.
+    options: EnumOptions,
+}
+
+impl<C: Context> Parse<C> for simple_extensions::EnumerationArg {
+    type Parsed = EnumerationArg;
+
+    type Error = ArgumentsItemError;
+
+    fn parse(self, ctx: &mut C) -> Result<EnumerationArg, ArgumentsItemError> {
+        Ok(EnumerationArg {
+            name: ArgumentsItem::parse_name(self.name)?,
+            description: ArgumentsItem::parse_description(self.description)?,
+            options: ctx.parse(self.options)?,
+        })
+    }
+}
+
+impl From<EnumerationArg> for simple_extensions::EnumerationArg {
+    fn from(value: EnumerationArg) -> Self {
+        simple_extensions::EnumerationArg {
+            name: value.name,
+            description: value.description,
+            options: value.options.into(),
+        }
+    }
+}
+
+impl From<EnumerationArg> for simple_extensions::ArgumentsItem {
+    fn from(value: EnumerationArg) -> Self {
+        simple_extensions::ArgumentsItem::EnumerationArg(value.into())
+    }
+}
+
+impl From<EnumerationArg> for ArgumentsItem {
+    fn from(value: EnumerationArg) -> Self {
+        ArgumentsItem::EnumArgument(value)
+    }
+}
+
+/// List of valid string options
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumOptions(Vec<String>);
+
+impl<C: Context> Parse<C> for simple_extensions::EnumOptions {
+    type Parsed = EnumOptions;
+
+    type Error = EnumOptionsError;
+
+    fn parse(self, _ctx: &mut C) -> Result<EnumOptions, EnumOptionsError> {
+        let options = self.0;
+        if options.is_empty() {
+            return Err(EnumOptionsError::EmptyList);
+        }
+
+        let mut unique_set = std::collections::HashSet::new();
+        for option in options.iter() {
+            if option.is_empty() {
+                return Err(EnumOptionsError::EmptyOption);
+            }
+            if !unique_set.insert(option) {
+                return Err(EnumOptionsError::DuplicatedOption(option.clone()));
+            }
+        }
+
+        Ok(EnumOptions(options))
+    }
+}
+
+impl From<EnumOptions> for simple_extensions::EnumOptions {
+    fn from(value: EnumOptions) -> Self {
+        simple_extensions::EnumOptions(value.0)
+    }
+}
+
+/// Parse errors for [simple_extensions::EnumOptions].
+#[derive(Debug, Error, PartialEq)]
+pub enum EnumOptionsError {
+    /// Empty list.
+    #[error("empty list")]
+    EmptyList,
+
+    /// Duplicated option.
+    #[error("duplicated option: {0}")]
+    DuplicatedOption(String),
+
+    /// Empty option.
+    #[error("empty option")]
+    EmptyOption,
+}
+
+/// Arguments that refer to a data value.
+#[derive(Clone, Debug)]
+pub struct ValueArg {
```

Comment:
Can you also add methods for this one?

Reply:
Added in [commit](https://github.com/substrait-io/substrait-rs/pull/185/commits/29f56d9136b6ff9db1044adb532c9e91a35071e0). The method for `value` is not added for now since it has not been parsed yet.